edx_rest_framework_extensions/auth/jwt/middleware.py

```diff
@@ -243,9 +259,43 @@ def process_view(self, request, view_func, view_args, view_kwargs):  # pylint: d
         else:
             attribute_value = 'missing-both'
             log.warning('Both JWT auth cookies missing. JWT auth cookies will not be reconstituted.')
+        monitoring.set_custom_attribute('request_jwt_cookie', attribute_value)
+
+    def _process_view_forgiving_jwt_cookies(self, request, view_func):
+        header_payload_cookie = request.COOKIES.get(jwt_cookie_header_payload_name())
+        signature_cookie = request.COOKIES.get(jwt_cookie_signature_name())
+
+        if header_payload_cookie and signature_cookie:
+            # Reconstitute JWT auth cookie if split cookies are available.
+            request.COOKIES[jwt_cookie_name()] = '{}{}{}'.format(
+                header_payload_cookie,
+                JWT_DELIMITER,
+                signature_cookie,
+            )
+            attribute_value = 'success'
+        elif header_payload_cookie or signature_cookie:
+            # Log unexpected case of only finding one cookie.
+            if not header_payload_cookie:
+                log_message, attribute_value = self._get_missing_cookie_message_and_attribute(
+                    jwt_cookie_header_payload_name()
+                )
+            if not signature_cookie:
+                log_message, attribute_value = self._get_missing_cookie_message_and_attribute(
+                    jwt_cookie_signature_name()
+                )
+            log.warning(log_message)
+        else:
+            attribute_value = 'missing-both'
+            log.warning('Both JWT auth cookies missing. JWT auth cookies will not be reconstituted.')
 
         monitoring.set_custom_attribute('request_jwt_cookie', attribute_value)
 
+        has_jwt_cookie = jwt_cookie_name() in request.COOKIES
+        is_set_request_user_for_jwt_cookie_enabled = get_setting(ENABLE_SET_REQUEST_USER_FOR_JWT_COOKIE)
+        if has_jwt_cookie and is_set_request_user_for_jwt_cookie_enabled:
```

Comment:
**Note to Reviewer:** It seems like this change would work even in the "original" case, and would potentially be more accurate. It could be moved to a separate PR (also with a rollout flag?), or just left here as part of the larger rollout.